projects/stader/index.js

```diff
@@ -1,86 +1,49 @@
-const { nullAddress } = require('../helper/unwrapLPs')
-const { fetchURL } = require("../helper/utils")
+const { fetchURL } = require("../helper/utils");
 
-async function hbarTvl(timestamp) {
-  const res = await fetchURL("https://universe.staderlabs.com/common/tvl")
-  return {
-    "hedera-hashgraph": res.data.hedera.native
-  }
-}
-
-async function maticTvl() {
-  const res = await fetchURL("https://universe.staderlabs.com/common/tvl")
-  return {
-    "matic-network": res.data.polygon.native
-  }
-}
+let commonTvlData = null;
 
-async function ftmTvl() {
-  const res = await fetchURL("https://universe.staderlabs.com/common/tvl")
-  return {
-    "fantom": res.data.fantom.native
+async function fetchCommonTvlData() {
+  if (!commonTvlData) {
+    const res = await fetchURL("https://universe.staderlabs.com/common/tvl");
+    commonTvlData = res.data;
   }
+  return commonTvlData;
 }
 
-async function terra2Tvl() {
-  const res = await fetchURL("https://universe.staderlabs.com/common/tvl")
-  return {
-    "terra-luna-2": res.data.terra.native
-  }
-}
-
-async function bscTvl() {
-  const res = await fetchURL("https://universe.staderlabs.com/common/tvl")
-  return {
-    "binancecoin": res.data.bnb.native
-  }
-}
+const createTvlFunction = (network, key) => {
+  return async () => {
+    const data = await fetchCommonTvlData();
+    let result = { [network]: data[key].native };
+    // matic is locked on ethereum network for Stader
+    if (network === "ethereum") {
+      result["matic-network"] = data["polygon"].native;
+    }
 
-async function nearTvl() {
-  const res = await fetchURL("https://universe.staderlabs.com/common/tvl")
-  return {
-    "near": res.data.near.native
-  }
-}
+    return result;
+  };
+};
 
 module.exports = {
   timetravel: false,
-  methodology: 'We aggregated the assets staked across Stader staking protocols',
-  /*terra: {
-    tvl,
-  },*/
+  methodology:
+    "We aggregated the assets staked across Stader staking protocols",
   hedera: {
-    tvl: hbarTvl,
+    tvl: createTvlFunction("hedera-hashgraph", "hedera"),
   },
-  // its on ethereum because funds are locked there
-  /* ethereum: {
-    tvl: maticTvl
-  }, */
   fantom: {
-    tvl: ftmTvl
+    tvl: createTvlFunction("fantom", "fantom"),
   },
   terra2: {
-    tvl: terra2Tvl
+    tvl: createTvlFunction("terra-luna-2", "terra"),
   },
   bsc: {
-    tvl: bscTvl
+    tvl: createTvlFunction("binancecoin", "bnb"),
   },
   near: {
-    tvl: nearTvl
+    tvl: createTvlFunction("near", "near"),
   },
   ethereum: {
-    tvl: async (_, _1, _2, { api }) => {
-
-      const res = await fetchURL("https://universe.staderlabs.com/common/tvl")
-      return {
-        "matic-network": res.data.polygon.native,
```

Comment:
assets on polygon are included here, else it would have caused a huge drop in tvl: https://defillama.com/protocol/stader.

I will leave this PR open for a bit while I turn everything on chain

Reply:
Cool. Let me know if you need any help.